### PR TITLE
refactor: spot trade order form

### PR DESCRIPTION
**Pull request by shane-moore**
## Changes
- Refactored spot's trade.vue into sub-components:
  - Market Type Selector
  - Order type selector
  - Order Inputs
  - Order Submit

The trade.vue component dropped from ~1700 lines to ~740!

### CODE REVIEW

File: components/partials/common/trade/advanced-settings/error.vue [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    slippageWarning(): string {
+      const { slippageTolerance, tradingTypeMarket } = this
+
+      if (!tradingTypeMarket) {
+        return ''
+      }
+
+      if (
+        new BigNumberInBase(slippageTolerance).gt(new BigNumberInBase(5)) &&
```
Comment (afterburn) [inline components/partials/common/trade/advanced-settings/error.vue:73]: Perhaps here you can put `new BigNumberInBase(slippageTolerance)` in a variable and reuse that as opposed to creating new BigNumbersInBase multiple times, more readable and less garbage collection

File: components/partials/common/trade/advanced-settings/index.vue [hunk 2]
```diff
@@ -241,6 +211,10 @@ export default Vue.extend({
       }
     },
 
+    handleHasErrorUpdateEvent(hasError: boolean): void {
+      this.$emit('update-has-advanced-settings-errors', hasError)
+    },
+
     handleReduceOnlyCheckboxToggle(checked: boolean) {
       this.$emit('set-reduce-only', checked)
```
Comment (bangjelkoski) [inline components/partials/common/trade/advanced-settings/index.vue:219]: Let's follow the pattern of `action:item` for the event names, here lets say we can have `update:reduceOnly` and above we have have `set:advancedSettingsErrors`

**Review by bangjelkoski — CHANGES_REQUESTED** [review]

File: components/partials/common/trade/advanced-settings/error.vue [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    },
+
+    tradingTypeMarket: {
+      type: Boolean,
+      required: true
+    }
+  },
+
+  computed: {
+    slippageError(): string | undefined {
```
Comment (ThomasRalee) [inline components/partials/common/trade/advanced-settings/error.vue:37]: Lets use `slippageTooHighError` directly and remove `slippageError` and `hasError` computed property

File: components/partials/common/trade/order-inputs.vue [hunk 4]
```diff
@@ -0,0 +1,535 @@
...
+      const { inputQuoteAmountToBigNumber, priceStep } = this
+
+      return (
+        !inputQuoteAmountToBigNumber.isNaN() &&
+        inputQuoteAmountToBigNumber.gt(0) &&
+        inputQuoteAmountToBigNumber.gte(priceStep)
+      )
+    },
+
+    priceStep(): string {
```
Comment (ThomasRalee) [inline components/partials/common/trade/order-inputs.vue:293]: Let's abstract `amountStep` to this component and abstract the way we format the placeholder values from the decimalPlaces into a function on the `formatter.ts` file.

File: components/partials/common/trade/order-inputs.vue [hunk 5]
```diff
@@ -0,0 +1,535 @@
...
+      this.$emit(
+        'update:has-advanced-settings-errors',
+        hasAdvancedSettingsErrors
+      )
+    },
+
+    updateProportionalPercentage(proportionalPercentage: number) {
+      this.inputProportionalPercentage = proportionalPercentage
+      this.$emit('update:proportional-percentage', proportionalPercentage)
+
```
Comment (ThomasRalee) [inline components/partials/common/trade/order-inputs.vue:369]: move the line break from 369 to 368

**Review by ThomasRalee — APPROVED** [review]

**Pull request merged**